### PR TITLE
Tbolt/3086 Date display discrepancy 

### LINKED ISSUE #3086: [Dev]Private Contractor Builder vs. Export view
There is a difference between the export view and the builder view. The export view shows the dates both in the review on screen and in the PDF but the builder view shows the text date not specified.

Related Information:
Browsers: Chrome, FireFox UserRoles:All applicable roles (excludes login error roles and fed admin)

Steps to Reproduce:
1. Navigate to any activities and proceed to the private contractor page.
2. Select the add a private contractor button and improperly add dates to the Contract start date field. For example 11/2019 or any invalid combo of MM/DD/YYYY (like 99/99/9999)

Expected Behavior/Result: Validation occurs on the date field and the user is asked to create a date in a certain format. Date is changed and shows as entered in both the builder and export view. 
Actual Results: If the user doesn't change the date to a valid date range, this produces on the builder view "date not specified". Entering into the edit area still shows the invalid dates. The invalid dates show on the export view. 
Unknown and TBD - what should the system behavior be when the user doesn't enter a valid date? 

Export (PDF VIEW)

User View(EAPD app)

This is done when 
- [ ] Design determines - what should the system behavior be when the user doesn't enter a valid date? 
- [ ] Dev implements that decision.

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #3086 

### This pull request changes...

- Changes validation logic on dates such that when an invalid date is entered it is still displayed in the review sections

### This is how to verify this change...
1. Checkout PR build / Login
2. Navigate to sections of the app that include date or date ranges
3. Verify the bug specified in the original ticket is fixed
4. Verify that invalid dates throughout the app are still displayed (a change was made to how we were showing dates - per parking lot convo, leave validation on the field as is. show "Invalid Date" on the places where the date is displayed.) 

### This pull request is ready to merge when...

- [x] Develop has updated automated tests (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] QA has manually tested and created bugs for any issues found (beyond minor fixes)
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [x] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [x] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/util/index.js [hunk 1]
```diff
@@ -127,11 +127,14 @@ export const generateKey = (() => {
  * @returns {String} Display-formatted date string.
  */
 export const stateDateToDisplay = date => {
+  if(!date) {
+    return 'Date not specified';
```
Comment (thetif) [inline web/src/util/index.js:131]: good thinking

**Review by thetif — APPROVED** [review]
looks good

**Comment by beparticular** [comment]
@tbolt per parking lot, leave validation on the field as is. show "Invalid Date" on the places where the date is displayed. 

@eStar386 and reviewers - when reviewing check other date fields in the app to make sure they are working as we've decided here.

File: web/src/util/index.js [hunk 2]
```diff
@@ -143,12 +146,10 @@ export const stateDateToDisplay = date => {
...
  */
 export const stateDateRangeToDisplay = (start, end) => {
-  const starty = moment(start, 'YYYY-MM-DD');
-  const endy = moment(end, 'YYYY-MM-DD');
-  if (starty.isValid() && endy.isValid()) {
-    return `${starty.format('M/D/YYYY')} - ${endy.format('M/D/YYYY')}`;
-  }
-  return 'Dates not specified';
+  const startDate = stateDateToDisplay(start);
+  const endDate = stateDateToDisplay(end);
```
Comment (thetif) [inline web/src/util/index.js:150]: nice reuse

**Review by thetif — APPROVED** [review]

**Review by beparticular — APPROVED** [review]
Looks great, thanks!

**Review by NAretakis — APPROVED** [review]
Good to go.

**Pull request merged**